### ISSUE TITLE
Issue when impersonating an user with \MarcinJozwikowski\EasyAdminPrettyUrls\Routing\PrettyUrlsResolver::resolveToPath

### ISSUE THREAD

**Issue by Geolim4** [marcin-jozwikowski/easy-admin-pretty-urls#45]
Hello,

I don't know why, but when I'm impersonating a user on Easy Admin this code is crashing in Easy Admin:
`{% if null == ea.userMenu.avatarUrl %}`

![image](https://github.com/user-attachments/assets/88433d6d-a062-45fd-9429-fddb14c22432)

`"?_switch_user=_exit"` is passed to `\MarcinJozwikowski\EasyAdminPrettyUrls\Routing\PrettyUrlsResolver::resolveToPath`
and since a path does not exists, it returns null. Can you add a string cast or nullable return ?

**Comment by Geolim4** [comment]
In fact the error is caused by:

\EasyCorp\Bundle\EasyAdminBundle\Factory\MenuFactory::generateMenuItemUrl
```php
        if (MenuItemDto::TYPE_EXIT_IMPERSONATION === $menuItemType) {
            // the switch parameter name can be changed, but this code assumes it's always
            // the default one because Symfony doesn't provide a generic exitImpersonationUrlGenerator
            return '?_switch_user=_exit';
        }


```

**Comment by Geolim4** [comment]
@marcin-jozwikowski an easy wuick way to fix that is to change the code as shown below:

```php
    public function resolveToPath(string $path): string
    {
        return parse_url($path, PHP_URL_PATH) ?? '/';
    }
```

**Comment by marcin-jozwikowski** [comment]
@Geolim4 Thanks again for doing the best QA I could dream of!

The issue has been fixed in https://github.com/marcin-jozwikowski/easy-admin-pretty-urls/releases/tag/v2.9.2 - I removed all menu items you cannot "stay on" from being considered. Logout page, user impersonation exit, section, and submenu are not actually links to valid pages. They are empty links or actions you're redirected from.
